ADMVEDU95/ADMVEDU95/Presentation/Home/Presenter/HomePresenter.swift

```diff
@@ -37,29 +40,31 @@ class HomePresenter: HomePresenterProtocol {
     func searchITunes(searchTerm: String, filter: String) {
         searchService?.searchResults(searchTerm: searchTerm,
                                     filter: filter) { result in
+            if !self.dataSource.isEmpty {
+                self.dataSource = []
+                self.view?.updateSearchResults()
+            }
             switch result {
             case .success(let response):
                 self.fetchDataFromResponse(response: response)
-                self.view?.successToGetData()
+                self.view?.updateSearchResults()
             case .failure(let error):
-                let alertMessage = self.getErrorAlertMessage(error: error)
-                self.view?.showAlert(title: "Error".localized(),
-                                     message: alertMessage)
+                let alertMessage = self.getErrorMessage(error: error)
+                self.handleRetry(message: alertMessage)
```

Comment:
there should be a global handler to check and handle error connection error, as we should not call this method at every presenter that has network requests